class com.adobe.creativesdk.foundation.internal.auth.d$a extends java.lang.Object
{
    final com.adobe.creativesdk.foundation.internal.auth.d a;
    private java.security.PrivateKey b;
    private java.security.PublicKey c;
    private boolean d;

    public void <init>(com.adobe.creativesdk.foundation.internal.auth.d)
    {
        com.adobe.creativesdk.foundation.internal.auth.d$a $r0;
        com.adobe.creativesdk.foundation.internal.auth.d $r1;
        boolean $z0;
        int varReplacer419;

        nop;

        $r0 := @this: com.adobe.creativesdk.foundation.internal.auth.d$a;

        $r1 := @parameter0: com.adobe.creativesdk.foundation.internal.auth.d;

        $r0.<com.adobe.creativesdk.foundation.internal.auth.d$a: com.adobe.creativesdk.foundation.internal.auth.d a> = $r1;

        specialinvoke $r0.<java.lang.Object: void <init>()>();

        varReplacer419 = 0;

        $r0.<com.adobe.creativesdk.foundation.internal.auth.d$a: boolean d> = varReplacer419;

        $z0 = $r0.<com.adobe.creativesdk.foundation.internal.auth.d$a: boolean d>;

        if $z0 != 0 goto label1;

        return;

     label1:
        specialinvoke $r0.<com.adobe.creativesdk.foundation.internal.auth.d$a: void b()>();

        return;
    }

    private void a()
    {
        com.adobe.creativesdk.foundation.internal.auth.d$a $r0;
        java.security.KeyStore $r1;
        java.security.KeyStore$Entry $r2;
        java.security.KeyStore$PrivateKeyEntry $r3;
        java.security.PrivateKey $r4;
        java.security.cert.Certificate $r5;
        java.security.PublicKey $r6;
        java.lang.Throwable $r7, $r11, $r12, $r13, $r14;
        com.adobe.creativesdk.foundation.internal.utils.logging.Level $r8;
        java.lang.String $r9, $r10, varReplacer421, varReplacer423;
        java.security.KeyStoreException r15;
        java.security.cert.CertificateException r16;
        java.security.NoSuchAlgorithmException r17;
        java.security.UnrecoverableEntryException r18;
        java.io.IOException r19;
        null_type varReplacer422, varReplacer424;

        nop;

        $r0 := @this: com.adobe.creativesdk.foundation.internal.auth.d$a;

     label01:
        varReplacer423 = "AndroidKeyStore";

        $r1 = staticinvoke <java.security.KeyStore: java.security.KeyStore getInstance(java.lang.String)>(varReplacer423);

        varReplacer424 = null;

        virtualinvoke $r1.<java.security.KeyStore: void load(java.security.KeyStore$LoadStoreParameter)>(varReplacer424);

        varReplacer421 = "adb_auth_foundation_rsa_keypair_alias";

        varReplacer422 = null;

        $r2 = virtualinvoke $r1.<java.security.KeyStore: java.security.KeyStore$Entry getEntry(java.lang.String,java.security.KeyStore$ProtectionParameter)>(varReplacer421, varReplacer422);

     label02:
        $r3 = (java.security.KeyStore$PrivateKeyEntry) $r2;

     label03:
        $r4 = virtualinvoke $r3.<java.security.KeyStore$PrivateKeyEntry: java.security.PrivateKey getPrivateKey()>();

     label04:
        $r0.<com.adobe.creativesdk.foundation.internal.auth.d$a: java.security.PrivateKey b> = $r4;

     label05:
        $r5 = virtualinvoke $r3.<java.security.KeyStore$PrivateKeyEntry: java.security.cert.Certificate getCertificate()>();

        $r6 = virtualinvoke $r5.<java.security.cert.Certificate: java.security.PublicKey getPublicKey()>();

     label06:
        $r0.<com.adobe.creativesdk.foundation.internal.auth.d$a: java.security.PublicKey c> = $r6;

        return;

     label07:
        $r7 := @caughtexception;

        $r8 = <com.adobe.creativesdk.foundation.internal.utils.logging.Level: com.adobe.creativesdk.foundation.internal.utils.logging.Level c>;

        $r9 = staticinvoke <com.adobe.creativesdk.foundation.internal.auth.d: java.lang.String e()>();

        r15 = (java.security.KeyStoreException) $r7;

        $r10 = virtualinvoke r15.<java.security.KeyStoreException: java.lang.String getMessage()>();

        staticinvoke <com.adobe.creativesdk.foundation.internal.utils.logging.a: void a(com.adobe.creativesdk.foundation.internal.utils.logging.Level,java.lang.String,java.lang.String,java.lang.Throwable)>($r8, $r9, $r10, $r7);

        return;

     label08:
        $r11 := @caughtexception;

        $r8 = <com.adobe.creativesdk.foundation.internal.utils.logging.Level: com.adobe.creativesdk.foundation.internal.utils.logging.Level c>;

        $r9 = staticinvoke <com.adobe.creativesdk.foundation.internal.auth.d: java.lang.String e()>();

        r16 = (java.security.cert.CertificateException) $r11;

        $r10 = virtualinvoke r16.<java.security.cert.CertificateException: java.lang.String getMessage()>();

        staticinvoke <com.adobe.creativesdk.foundation.internal.utils.logging.a: void a(com.adobe.creativesdk.foundation.internal.utils.logging.Level,java.lang.String,java.lang.String,java.lang.Throwable)>($r8, $r9, $r10, $r11);

        return;

     label09:
        $r12 := @caughtexception;

        $r8 = <com.adobe.creativesdk.foundation.internal.utils.logging.Level: com.adobe.creativesdk.foundation.internal.utils.logging.Level c>;

        $r9 = staticinvoke <com.adobe.creativesdk.foundation.internal.auth.d: java.lang.String e()>();

        r17 = (java.security.NoSuchAlgorithmException) $r12;

        $r10 = virtualinvoke r17.<java.security.NoSuchAlgorithmException: java.lang.String getMessage()>();

        staticinvoke <com.adobe.creativesdk.foundation.internal.utils.logging.a: void a(com.adobe.creativesdk.foundation.internal.utils.logging.Level,java.lang.String,java.lang.String,java.lang.Throwable)>($r8, $r9, $r10, $r12);

        return;

     label10:
        $r13 := @caughtexception;

        $r8 = <com.adobe.creativesdk.foundation.internal.utils.logging.Level: com.adobe.creativesdk.foundation.internal.utils.logging.Level c>;

        $r9 = staticinvoke <com.adobe.creativesdk.foundation.internal.auth.d: java.lang.String e()>();

        r18 = (java.security.UnrecoverableEntryException) $r13;

        $r10 = virtualinvoke r18.<java.security.UnrecoverableEntryException: java.lang.String getMessage()>();

        staticinvoke <com.adobe.creativesdk.foundation.internal.utils.logging.a: void a(com.adobe.creativesdk.foundation.internal.utils.logging.Level,java.lang.String,java.lang.String,java.lang.Throwable)>($r8, $r9, $r10, $r13);

        return;

     label11:
        $r14 := @caughtexception;

        $r8 = <com.adobe.creativesdk.foundation.internal.utils.logging.Level: com.adobe.creativesdk.foundation.internal.utils.logging.Level c>;

        $r9 = staticinvoke <com.adobe.creativesdk.foundation.internal.auth.d: java.lang.String e()>();

        r19 = (java.io.IOException) $r14;

        $r10 = virtualinvoke r19.<java.io.IOException: java.lang.String getMessage()>();

        staticinvoke <com.adobe.creativesdk.foundation.internal.utils.logging.a: void a(com.adobe.creativesdk.foundation.internal.utils.logging.Level,java.lang.String,java.lang.String,java.lang.Throwable)>($r8, $r9, $r10, $r14);

        return;

        catch java.security.KeyStoreException from label01 to label02 with label07;
        catch java.security.KeyStoreException from label03 to label04 with label07;
        catch java.security.KeyStoreException from label05 to label06 with label07;
        catch java.security.cert.CertificateException from label01 to label02 with label08;
        catch java.security.cert.CertificateException from label03 to label04 with label08;
        catch java.security.cert.CertificateException from label05 to label06 with label08;
        catch java.security.NoSuchAlgorithmException from label01 to label02 with label09;
        catch java.security.NoSuchAlgorithmException from label03 to label04 with label09;
        catch java.security.NoSuchAlgorithmException from label05 to label06 with label09;
        catch java.security.UnrecoverableEntryException from label01 to label02 with label10;
        catch java.security.UnrecoverableEntryException from label03 to label04 with label10;
        catch java.security.UnrecoverableEntryException from label05 to label06 with label10;
        catch java.io.IOException from label01 to label02 with label11;
        catch java.io.IOException from label03 to label04 with label11;
        catch java.io.IOException from label05 to label06 with label11;
    }

    private void b()
    {
        com.adobe.creativesdk.foundation.internal.auth.d$a $r0;
        java.lang.String $r1, varReplacer420;
        java.security.PrivateKey $r2;
        java.security.PublicKey $r3;
        boolean $z0;
        com.adobe.creativesdk.foundation.internal.auth.d $r4;
        java.lang.IllegalArgumentException $r5;

        nop;

        $r0 := @this: com.adobe.creativesdk.foundation.internal.auth.d$a;

        specialinvoke $r0.<com.adobe.creativesdk.foundation.internal.auth.d$a: void a()>();

        $r2 = $r0.<com.adobe.creativesdk.foundation.internal.auth.d$a: java.security.PrivateKey b>;

        if $r2 == null goto label1;

        $r3 = $r0.<com.adobe.creativesdk.foundation.internal.auth.d$a: java.security.PublicKey c>;

        if $r3 != null goto label3;

     label1:
        $z0 = 1;

     label2:
        if $z0 == 0 goto label5;

        $r4 = $r0.<com.adobe.creativesdk.foundation.internal.auth.d$a: com.adobe.creativesdk.foundation.internal.auth.d a>;

        $r1 = staticinvoke <com.adobe.creativesdk.foundation.internal.auth.d: java.lang.String a(com.adobe.creativesdk.foundation.internal.auth.d)>($r4);

        if $r1 == null goto label4;

        $r5 = new java.lang.IllegalArgumentException;

        varReplacer420 = "Auth Foundation :AESKeyMgr Failed to initialize Private keys - but already have a valid aes key stored !!";

        specialinvoke $r5.<java.lang.IllegalArgumentException: void <init>(java.lang.String)>(varReplacer420);

        throw $r5;

     label3:
        $z0 = 0;

        goto label2;

     label4:
        specialinvoke $r0.<com.adobe.creativesdk.foundation.internal.auth.d$a: void d()>();

        specialinvoke $r0.<com.adobe.creativesdk.foundation.internal.auth.d$a: void a()>();

     label5:
        return;
    }

    private javax.crypto.Cipher c()
    {
        com.adobe.creativesdk.foundation.internal.auth.d$a $r0;
        javax.crypto.Cipher $r1;
        java.lang.Exception $r2;
        java.lang.IllegalArgumentException $r3;
        java.lang.String varReplacer8265, varReplacer8266, varReplacer8267;

        nop;

        $r0 := @this: com.adobe.creativesdk.foundation.internal.auth.d$a;

     label1:
        varReplacer8265 = "RSA/ECB/PKCS1Padding";

        varReplacer8266 = "AndroidOpenSSL";

        $r1 = staticinvoke <javax.crypto.Cipher: javax.crypto.Cipher getInstance(java.lang.String,java.lang.String)>(varReplacer8265, varReplacer8266);

     label2:
        return $r1;

     label3:
        $r2 := @caughtexception;

        $r3 = new java.lang.IllegalArgumentException;

        varReplacer8267 = "Auth Foundation : Auth AESKeyMgr : Failed to create Cipher";

        specialinvoke $r3.<java.lang.IllegalArgumentException: void <init>(java.lang.String,java.lang.Throwable)>(varReplacer8267, $r2);

        throw $r3;

        catch java.lang.Exception from label1 to label2 with label3;
    }

    private byte[] c(byte[])
    {
        com.adobe.creativesdk.foundation.internal.auth.d$a $r0;
        byte[] $r1;
        javax.crypto.Cipher $r2;
        java.security.PublicKey $r3;
        java.lang.Exception $r4;
        com.adobe.creativesdk.foundation.internal.utils.logging.Level $r5;
        java.lang.String $r6, $r7, varReplacer8269;
        java.lang.IllegalArgumentException $r8;
        int varReplacer8268;

        nop;

        $r0 := @this: com.adobe.creativesdk.foundation.internal.auth.d$a;

        $r1 := @parameter0: byte[];

        $r2 = specialinvoke $r0.<com.adobe.creativesdk.foundation.internal.auth.d$a: javax.crypto.Cipher c()>();

     label1:
        $r3 = $r0.<com.adobe.creativesdk.foundation.internal.auth.d$a: java.security.PublicKey c>;

        varReplacer8268 = 1;

        virtualinvoke $r2.<javax.crypto.Cipher: void init(int,java.security.Key)>(varReplacer8268, $r3);

        $r1 = virtualinvoke $r2.<javax.crypto.Cipher: byte[] doFinal(byte[])>($r1);

     label2:
        return $r1;

     label3:
        $r4 := @caughtexception;

        $r5 = <com.adobe.creativesdk.foundation.internal.utils.logging.Level: com.adobe.creativesdk.foundation.internal.utils.logging.Level c>;

        $r6 = staticinvoke <com.adobe.creativesdk.foundation.internal.auth.d: java.lang.String e()>();

        $r7 = virtualinvoke $r4.<java.lang.Exception: java.lang.String getMessage()>();

        staticinvoke <com.adobe.creativesdk.foundation.internal.utils.logging.a: void a(com.adobe.creativesdk.foundation.internal.utils.logging.Level,java.lang.String,java.lang.String,java.lang.Throwable)>($r5, $r6, $r7, $r4);

        $r8 = new java.lang.IllegalArgumentException;

        varReplacer8269 = "Auth Foundation : Auth AESKeyMgr : Failed to encrypt using public key";

        specialinvoke $r8.<java.lang.IllegalArgumentException: void <init>(java.lang.String,java.lang.Throwable)>(varReplacer8269, $r4);

        throw $r8;

        catch java.lang.Exception from label1 to label2 with label3;
    }

    private void d()
    {
        com.adobe.creativesdk.foundation.internal.auth.d$a $r0;
        java.util.Calendar $r1;
        java.util.Date $r2, $r3;
        java.security.KeyPairGenerator $r4;
        com.adobe.creativesdk.foundation.internal.c.a $r5;
        android.content.Context $r6;
        android.security.KeyPairGeneratorSpec$Builder $r7;
        java.math.BigInteger $r8;
        javax.security.auth.x500.X500Principal $r9;
        java.lang.Object[] $r10;
        android.content.pm.ApplicationInfo $r11;
        java.lang.String $r12, $r16, varReplacer425, varReplacer426, varReplacer429, varReplacer430, varReplacer431;
        int $i0, varReplacer427, varReplacer428, varReplacer432;
        android.security.KeyPairGeneratorSpec $r13;
        java.lang.Exception $r14;
        com.adobe.creativesdk.foundation.internal.utils.logging.Level $r15;
        java.lang.IllegalArgumentException $r17;

        nop;

        $r0 := @this: com.adobe.creativesdk.foundation.internal.auth.d$a;

        $r1 = staticinvoke <java.util.Calendar: java.util.Calendar getInstance()>();

        $r2 = virtualinvoke $r1.<java.util.Calendar: java.util.Date getTime()>();

        varReplacer427 = 1;

        varReplacer428 = 25;

        virtualinvoke $r1.<java.util.Calendar: void add(int,int)>(varReplacer427, varReplacer428);

        $r3 = virtualinvoke $r1.<java.util.Calendar: java.util.Date getTime()>();

     label01:
        varReplacer430 = "RSA";

        varReplacer431 = "AndroidKeyStore";

        $r4 = staticinvoke <java.security.KeyPairGenerator: java.security.KeyPairGenerator getInstance(java.lang.String,java.lang.String)>(varReplacer430, varReplacer431);

        $r5 = staticinvoke <com.adobe.creativesdk.foundation.internal.c.a: com.adobe.creativesdk.foundation.internal.c.a a()>();

        $r6 = virtualinvoke $r5.<com.adobe.creativesdk.foundation.internal.c.a: android.content.Context b()>();

     label02:
        $r7 = new android.security.KeyPairGeneratorSpec$Builder;

     label03:
        specialinvoke $r7.<android.security.KeyPairGeneratorSpec$Builder: void <init>(android.content.Context)>($r6);

        varReplacer429 = "adb_auth_foundation_rsa_keypair_alias";

        $r7 = virtualinvoke $r7.<android.security.KeyPairGeneratorSpec$Builder: android.security.KeyPairGeneratorSpec$Builder setAlias(java.lang.String)>(varReplacer429);

        $r7 = virtualinvoke $r7.<android.security.KeyPairGeneratorSpec$Builder: android.security.KeyPairGeneratorSpec$Builder setStartDate(java.util.Date)>($r2);

        $r7 = virtualinvoke $r7.<android.security.KeyPairGeneratorSpec$Builder: android.security.KeyPairGeneratorSpec$Builder setEndDate(java.util.Date)>($r3);

        $r8 = staticinvoke <java.math.BigInteger: java.math.BigInteger valueOf(long)>(1L);

        $r7 = virtualinvoke $r7.<android.security.KeyPairGeneratorSpec$Builder: android.security.KeyPairGeneratorSpec$Builder setSerialNumber(java.math.BigInteger)>($r8);

     label04:
        $r9 = new javax.security.auth.x500.X500Principal;

     label05:
        $r10 = newarray (java.lang.Object)[2];

        $r10[0] = "adb_auth_foundation_rsa_keypair_alias";

        $r11 = virtualinvoke $r6.<android.content.Context: android.content.pm.ApplicationInfo getApplicationInfo()>();

        $r12 = $r11.<android.content.pm.ApplicationInfo: java.lang.String packageName>;

        $r10[1] = $r12;

        varReplacer425 = "CN=%s, OU=%s";

        $r12 = staticinvoke <java.lang.String: java.lang.String format(java.lang.String,java.lang.Object[])>(varReplacer425, $r10);

        specialinvoke $r9.<javax.security.auth.x500.X500Principal: void <init>(java.lang.String)>($r12);

        $r7 = virtualinvoke $r7.<android.security.KeyPairGeneratorSpec$Builder: android.security.KeyPairGeneratorSpec$Builder setSubject(javax.security.auth.x500.X500Principal)>($r9);

     label06:
        $i0 = <android.os.Build$VERSION: int SDK_INT>;

        if $i0 < 19 goto label08;

     label07:
        varReplacer432 = 2048;

        virtualinvoke $r7.<android.security.KeyPairGeneratorSpec$Builder: android.security.KeyPairGeneratorSpec$Builder setKeySize(int)>(varReplacer432);

     label08:
        $r13 = virtualinvoke $r7.<android.security.KeyPairGeneratorSpec$Builder: android.security.KeyPairGeneratorSpec build()>();

        virtualinvoke $r4.<java.security.KeyPairGenerator: void initialize(java.security.spec.AlgorithmParameterSpec)>($r13);

        virtualinvoke $r4.<java.security.KeyPairGenerator: java.security.KeyPair generateKeyPair()>();

     label09:
        return;

     label10:
        $r14 := @caughtexception;

        $r15 = <com.adobe.creativesdk.foundation.internal.utils.logging.Level: com.adobe.creativesdk.foundation.internal.utils.logging.Level c>;

        $r12 = staticinvoke <com.adobe.creativesdk.foundation.internal.auth.d: java.lang.String e()>();

        $r16 = virtualinvoke $r14.<java.lang.Exception: java.lang.String getMessage()>();

        staticinvoke <com.adobe.creativesdk.foundation.internal.utils.logging.a: void a(com.adobe.creativesdk.foundation.internal.utils.logging.Level,java.lang.String,java.lang.String,java.lang.Throwable)>($r15, $r12, $r16, $r14);

        $r17 = new java.lang.IllegalArgumentException;

        varReplacer426 = "Failed to generate RSA Public Private Key pair";

        specialinvoke $r17.<java.lang.IllegalArgumentException: void <init>(java.lang.String)>(varReplacer426);

        throw $r17;

        catch java.lang.Exception from label01 to label02 with label10;
        catch java.lang.Exception from label03 to label04 with label10;
        catch java.lang.Exception from label05 to label06 with label10;
        catch java.lang.Exception from label07 to label09 with label10;
    }

    private byte[] d(byte[])
    {
        com.adobe.creativesdk.foundation.internal.auth.d$a $r0;
        byte[] $r1;
        javax.crypto.Cipher $r2;
        java.security.PrivateKey $r3;
        java.lang.Exception $r4;
        com.adobe.creativesdk.foundation.internal.utils.logging.Level $r5;
        java.lang.String $r6, $r7, varReplacer8263;
        java.lang.IllegalArgumentException $r8;
        int varReplacer8264;

        nop;

        $r0 := @this: com.adobe.creativesdk.foundation.internal.auth.d$a;

        $r1 := @parameter0: byte[];

        $r2 = specialinvoke $r0.<com.adobe.creativesdk.foundation.internal.auth.d$a: javax.crypto.Cipher c()>();

     label1:
        $r3 = $r0.<com.adobe.creativesdk.foundation.internal.auth.d$a: java.security.PrivateKey b>;

        varReplacer8264 = 2;

        virtualinvoke $r2.<javax.crypto.Cipher: void init(int,java.security.Key)>(varReplacer8264, $r3);

        $r1 = virtualinvoke $r2.<javax.crypto.Cipher: byte[] doFinal(byte[])>($r1);

     label2:
        return $r1;

     label3:
        $r4 := @caughtexception;

        $r5 = <com.adobe.creativesdk.foundation.internal.utils.logging.Level: com.adobe.creativesdk.foundation.internal.utils.logging.Level c>;

        $r6 = staticinvoke <com.adobe.creativesdk.foundation.internal.auth.d: java.lang.String e()>();

        $r7 = virtualinvoke $r4.<java.lang.Exception: java.lang.String getMessage()>();

        staticinvoke <com.adobe.creativesdk.foundation.internal.utils.logging.a: void a(com.adobe.creativesdk.foundation.internal.utils.logging.Level,java.lang.String,java.lang.String,java.lang.Throwable)>($r5, $r6, $r7, $r4);

        $r8 = new java.lang.IllegalArgumentException;

        varReplacer8263 = "Auth Foundation : Auth AESKeyMgr : Failed to decyrpt using private key";

        specialinvoke $r8.<java.lang.IllegalArgumentException: void <init>(java.lang.String,java.lang.Throwable)>(varReplacer8263, $r4);

        throw $r8;

        catch java.lang.Exception from label1 to label2 with label3;
    }

    public byte[] a(byte[])
    {
        com.adobe.creativesdk.foundation.internal.auth.d$a $r0;
        boolean $z0;
        byte[] $r1;

        nop;

        $r0 := @this: com.adobe.creativesdk.foundation.internal.auth.d$a;

        $r1 := @parameter0: byte[];

        $z0 = $r0.<com.adobe.creativesdk.foundation.internal.auth.d$a: boolean d>;

        if $z0 != 0 goto label1;

        return $r1;

     label1:
        $r1 = specialinvoke $r0.<com.adobe.creativesdk.foundation.internal.auth.d$a: byte[] c(byte[])>($r1);

        return $r1;
    }

    public byte[] b(byte[])
    {
        com.adobe.creativesdk.foundation.internal.auth.d$a $r0;
        boolean $z0;
        byte[] $r1;

        nop;

        $r0 := @this: com.adobe.creativesdk.foundation.internal.auth.d$a;

        $r1 := @parameter0: byte[];

        $z0 = $r0.<com.adobe.creativesdk.foundation.internal.auth.d$a: boolean d>;

        if $z0 != 0 goto label1;

        return $r1;

     label1:
        $r1 = specialinvoke $r0.<com.adobe.creativesdk.foundation.internal.auth.d$a: byte[] d(byte[])>($r1);

        return $r1;
    }
}
